src/app.rs

```diff
@@ -1717,6 +1724,28 @@ impl Application for App {
                 }
                 return self.update_notification();
             }
+            Message::Refresh(entity_opt) => {
+                let entity = entity_opt.unwrap_or_else(|| self.tab_model.active());
+                if let Some(tab) = self.tab_model.data_mut::<Tab>(entity) {
+                    let location = tab.location.clone();
+                    tab.set_items(vec![]);
```

Comment:
Essentially, I set an "empty" value for the given tab and register a delayed callback that will either rescan or search again. Maybe something better could be done here.